Adjust search mingw32 lib search to support llvm-mingw32 compiler.

RussellHaley opened the pull request:
Hello, 

As per #1135, I was able to get the llvm-mingw32 toolchain to work with luarocks. I'm not sure why the mingw32 internal build script doesn't respect variables.LUALIB from the config file, but the fix is minor. I simply adjusted the search path as provided by cfg.lua. I moved the `?.lib` search string to the front of the table definition. I don't expect this to affect mingw32-gcc, but that's a guess (based on an old faded memory) on my part. I suspect it either doesn't generate lib files (so won't find it in a search string) or is agnostic of their use. I'm indifferent to leaving the comment in (e.g. take it out if you feel). I thought it appropriate given there was an opposing comment there before.

hishamhm committed:
speed up external dependency check

hishamhm committed:
build: do not download sources when building only deps

hishamhm committed:
appveyor: bump Python to 3.7

RussellHaley committed:
Adjust search mingw32 lib search patterns to support llvm-mingw32 compiler toolchain.

hishamhm committed:
Windows: force .lib for LUALIB

hishamhm committed:
tests: drop LuaSec dependency on tests, use fixtures instead

hishamhm commented on src/luarocks/core/cfg.lua:
```diff
@@ -347,8 +347,10 @@ local function make_defaults(lua_version, target_cpu, platforms, home)
+         --~ 2020-01-23: putting ?.lib first allows use of the llvm-mingw32 compiler toolchain
+         --~ https://github.com/mstorsjo/llvm-mingw
+         --~ The LLVM linker doesn't support linking to a dll (GNU ld does).         
+         lib = { "?.lib", "lib?.dll.a", "?.dll.a", "lib?.a", "cyg?.dll", "lib?.dll", "?.dll" },
```
Is there a way to check that this change does not affect non-LLVM mingw32 in some way? Not sure if the testsuite (and the CI run for `spec/build_spec.lua` in Appveyor) covers it.

hishamhm replied:
> Not sure if the testsuite (and the CI run for spec/build_spec.lua in Appveyor) covers it.

I think there may be a chance it does, in the very latest code of `master`. You might want to rebase this PR over the latest `origin/master` and force-push it so we get a new CI run.

hishamhm replied:
On deeper inspection, I don't think the test suite actually checks that the build of a module with an external dependency works correctly (i.e. that it produces a module that the Lua interpreter is really able to load and use). I'm reluctant about merging this before further testing, I can try my hand at extending the tests (probably next week).

RussellHaley replied:
I got a build working last night so I can move forward and test; no rush on up-streaming it. What platform are you concerned about? Windows, Linux?

hishamhm commented:
The other platform that would be affected by this change would be mingw
with gcc.

I recently added a test that builds a dummy external library on the spot
(look for "fixturedep") and uses it as an external dependency, but the test
doesn't go as far as verifying that the linkage was correct (i.e. that the
Lua module is actually able to use the C library successfully).

RussellHaley committed:
Build file updates

- .gitignore - updated to include windows binary build outputs.
- all_in_one - changed default path for external dependencies (e.g. mingw)
- binary/lua-bz2-0.1.0-1.rockspec - changed protocol for anonymous download
